.github/workflows/ci.yml

```diff
@@ -133,10 +133,22 @@ jobs:
         shell: bash
         run: echo ::set-output name=flavor::$([[ "${{ matrix.os }}" == "cuda" ]] && echo "Cuda" || echo "CPU")
 
+      # Clean tests (kernel caching disabled) use more CPU and memory, so we only run them for
+      # scheduled runs, pushes to master, and pushes to tags.
+      # Parallelization also needs to be disabled to avoid Out of Memory errors.
+      - name: Set environment variables
+        id: env
+        shell: bash
+        run: |
+          jsonValue="{}"
+          if [[ ${{ github.event_name == 'schedule' || (github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))) }} ]]; then
+            jsonValue="{\"ILGPU_CLEAN_TESTS\":\"true\", \"ILGPU_TEST_PARALLELIZATION_LEVEL\":\"1\"}"
```

Comment:
@ljubon latest discussion on the topic is to _not set_ `ILGPU_TEST_PARALLELIZATION_LEVEL`. So this line should be changed to:
```
jsonValue="{\"ILGPU_CLEAN_TESTS\":\"true\"}"
```

Reply:
This should work after we merge #741.

Reply:
This change would essentially become a no-op compared to what we already have today:
https://github.com/m4rs-mt/ILGPU/blob/3e59b1a1c523eb501405e392dc5bd5058e4562ee/.github/workflows/ci.yml#L138-L139

Now that #741 has been merged, I would suggest to close this PR instead. Sorry @ljubon! 😞

Reply:
@jgiannuzzi Do you think it's not worth keeping this PR and its changes for future CI pipeline changes/updates when you push to `master`?

Reply:
@m4rs-mt I'd suggest we reopen this PR if the need arises (and adapt if necessary)

Reply:
hey guys, as mentioned I've pushed the latest change which we originally wanted
if you all agree, I would proceed with the closure of this PR (as is) just in case we want in future to re-open or add additional work on this PR